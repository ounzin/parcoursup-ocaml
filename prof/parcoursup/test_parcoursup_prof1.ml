(*
fonctionnalité testée:

"""
    Attention, on doit pouvoir travailler avec plusieurs sessions simultanément
    sans qu'il y ait d'interférences, par exemple une session "parcoursup belge 2030" 
    et une autre session "parcoursup français 2030".
"""

*)

open Test_parcoursup_utils

let ajoute_candidat i session ~nom_candidat =
  Format.printf "[SESSION %d] " i;
  ajoute_candidat session ~nom_candidat

let ajoute_formation i session ~nom_formation ~capacite =
  Format.printf "[SESSION %d] " i;
  ajoute_formation session ~nom_formation ~capacite

let ajoute_commission i session ~nom_formation tab =
  Format.printf "[SESSION %d] " i;
  ajoute_commission session ~nom_formation tab

let ajoute_voeu i session ~rang_repondeur ~nom_candidat ~nom_formation =
  Format.printf "[SESSION %d] " i;
  ajoute_voeu session ~rang_repondeur ~nom_candidat ~nom_formation

let affiche_voeux_en_attente i session s =
  Format.printf "[SESSION %d] " i;
  affiche_voeux_en_attente session s

let affiche_propositions_en_attente i session s =
  Format.printf "[SESSION %d] " i;
  affiche_propositions_en_attente session s
  
let session = Array.init 4 (fun _ -> nouvelle_session ()) 

let () = 
  for i = 0 to 1 do
    ajoute_candidat i session.(i) ~nom_candidat:"a";
    ajoute_formation i session.(i) ~nom_formation:"A" ~capacite:1;
    ajoute_formation i session.(i) ~nom_formation:"B" ~capacite:1;
    ajoute_commission i session.(i) ~nom_formation:"A" ["a"]; 
    ajoute_commission i session.(i) ~nom_formation:"B" ["a"]
  done;
  ajoute_voeu 0 session.(0) ~rang_repondeur:(Some 1) ~nom_candidat:"a" ~nom_formation:"A";
  ajoute_voeu 0 session.(0) ~rang_repondeur:(Some 2) ~nom_candidat:"a" ~nom_formation:"B";
  ajoute_voeu 1 session.(1) ~rang_repondeur:(Some 1) ~nom_candidat:"a" ~nom_formation:"B";
  ajoute_voeu 1 session.(1) ~rang_repondeur:(Some 2) ~nom_candidat:"a" ~nom_formation:"A";
  for i = 0 to 1 do
    reunit_commissions session.(i);
    nouveau_jour session.(i);
    affiche_voeux_en_attente i session.(i) "a";
    affiche_propositions_en_attente i session.(i) "a"
  done;
  for i = 2 to 3 do
    ajoute_candidat i session.(i) ~nom_candidat:"a";
    ajoute_candidat i session.(i) ~nom_candidat:"b";
    ajoute_formation i session.(i) ~nom_formation:"A" ~capacite:1;
    ajoute_voeu i session.(i) ~rang_repondeur:None ~nom_candidat:"a" ~nom_formation:"A";
    ajoute_voeu i session.(i) ~rang_repondeur:None ~nom_candidat:"b" ~nom_formation:"A";
  done;
  ajoute_commission 2 session.(2) ~nom_formation:"A" ["a";"b"]; 
  ajoute_commission 3 session.(3) ~nom_formation:"A" ["b";"a"]; 
  for i = 2 to 3 do
    reunit_commissions session.(i);
    nouveau_jour session.(i);
    affiche_voeux_en_attente i session.(i) "a";
    affiche_voeux_en_attente i session.(i) "b";
  done;
