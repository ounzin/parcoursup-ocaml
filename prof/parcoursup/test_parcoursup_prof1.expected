nouvelle session
nouvelle session
nouvelle session
nouvelle session
[SESSION 0] ajoute candidat a
[SESSION 0] ajoute formation A (1 places)
[SESSION 0] ajoute formation B (1 places)
[SESSION 0] commission de classement de A : [a]
[SESSION 0] commission de classement de B : [a]
[SESSION 1] ajoute candidat a
[SESSION 1] ajoute formation A (1 places)
[SESSION 1] ajoute formation B (1 places)
[SESSION 1] commission de classement de A : [a]
[SESSION 1] commission de classement de B : [a]
[SESSION 0] ajoute voeu de a pour A (rang=1)
[SESSION 0] ajoute voeu de a pour B (rang=2)
[SESSION 1] ajoute voeu de a pour B (rang=1)
[SESSION 1] ajoute voeu de a pour A (rang=2)
réunit commissions
nouveau jour
[SESSION 0] Voeux en attente de a : {}
[SESSION 0] Propositions en attente de a : {A}
réunit commissions
nouveau jour
[SESSION 1] Voeux en attente de a : {}
[SESSION 1] Propositions en attente de a : {B}
[SESSION 2] ajoute candidat a
[SESSION 2] ajoute candidat b
[SESSION 2] ajoute formation A (1 places)
[SESSION 2] ajoute voeu de a pour A
[SESSION 2] ajoute voeu de b pour A
[SESSION 3] ajoute candidat a
[SESSION 3] ajoute candidat b
[SESSION 3] ajoute formation A (1 places)
[SESSION 3] ajoute voeu de a pour A
[SESSION 3] ajoute voeu de b pour A
[SESSION 2] commission de classement de A : [a; b]
[SESSION 3] commission de classement de A : [b; a]
réunit commissions
nouveau jour
[SESSION 2] Voeux en attente de a : {}
[SESSION 2] Voeux en attente de b : {A(0)}
réunit commissions
nouveau jour
[SESSION 3] Voeux en attente de a : {A(0)}
[SESSION 3] Voeux en attente de b : {}
